source/common/network/lc_trie.h

```diff
@@ -531,18 +525,23 @@ std::string LcTrie::LcTrieInternal<IpType, address_size>::getTag(const IpType& i
     address = node.address_;
   }
 
-  // /0 will match all IP addresses.
-  if (ip_prefixes_[address].length_ == 0) {
-    return ip_prefixes_[address].tag_;
-  }
-
-  // Check the input IP address is within the CIDR range stored in ip_prefixes_[adr] by XOR'ing the
-  // two values and checking that up until the length of the CIDR range the value is 0.
-  IpType bitmask = ip_prefixes_[address].ip_ ^ ip_address;
-  if (extractBits<IpType, address_size>(0, ip_prefixes_[address].length_, bitmask) == 0) {
-    return ip_prefixes_[address].tag_;
+  // The prefix table entry ip_prefixes_[address] contains either a single prefix or
```

Comment:
Ok, added! With this PR, the code is relatively clean, because the /0 special case is encapsulated in the `extractBits` function, but the drawback is that the extra conditional in that function will make the common case (prefixes of nonzero length) a bit slower.